core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -460,7 +463,15 @@ public ByteBuffer keyMetadata() {
 
   @Override
   public List<Long> splitOffsets() {
-    return ArrayUtil.toLongList(splitOffsets);
+    if (splitOffsetsAsList == null && splitOffsets != null) {
```

Comment:
Is second null check required ?  Looks like the internal method handles null, and original code did not have it.